dex-it/src/test/scala/com/wavesplatform/it/sync/OrderHistoryTestSuite.scala

```diff
@@ -1,95 +1,84 @@
-
-  val customDB       = "user_db"
-  val customUser     = "user"
-  val customPassword = "user"
-
-  val postgresImageName           = "postgres"
-  val postgresContainerName       = "pgc"
-  val postgresContainerPort       = "5432"
-  val postgresContainerIp: String = dockerClient.ipForNode(10)
-
-  val postgresContainerLauncher =
-    new DockerContainerLauncher(
-      imageName = postgresImageName,
-      containerName = postgresContainerName,
-      containerIp = postgresContainerIp,
-      containerPort = postgresContainerPort,
-      imageTag = "10",
-      env = List(s"POSTGRES_DB=$customDB", s"POSTGRES_USER=$customUser", s"POSTGRES_PASSWORD=$customPassword"),
-      networkName = dockerClient.network().name
-    )
-
-  private val batchLingerMs: Int = OrderHistorySettings.defaultBatchLingerMs
+import scala.util.Try
 
-  private def getPostgresContainerHostPort: String = postgresContainerLauncher.getHostPort.explicitGet()
+class OrderHistoryTestSuite extends NewMatcherSuiteBase {
 
-  // TODO rewrite
-  @annotation.tailrec
-  private def retry[T](attemptsCount: Int, delayInMs: Int)(fn: => T): Try[T] = {
-    Try { fn } match {
-      case Failure(_) if attemptsCount > 1 => if (delayInMs != 0) Thread.sleep(delayInMs); retry(attemptsCount - 1, delayInMs)(fn)
-      case Failure(ex)                     => throw ex
-      case success                         => success
-    }
-  }
+  private val customDB       = "user_db"
+  private val customUser     = "user"
+  private val customPassword = "user"
 
-  def getFileContentStr(fileName: String): String = {
-    val fileStream = getClass.getResourceAsStream(fileName)
-    Source.fromInputStream(fileStream).getLines.toSeq.mkString
-  }
+  private val postgresContainerName = "pgc"
+  private val postgresContainerPort = 5432
 
-  def createDatabase(): Unit = {
+  override protected val dexInitialSuiteConfig: Config = ConfigFactory.parseString(
+    s"""
+       |waves.dex {
+       |  price-assets = [ "$UsdId", "$BtcId", "WAVES", "$EthId", "$WctId" ]
+       |  ${getPostgresConnectionCfgString(postgresContainerName, postgresContainerPort)}
+       |  order-history.enabled = yes
+       |}
+    """.stripMargin
+  )
 
-    val createDatabaseFileName    = "init-user-db.sh"
-    val createDatabaseFileContent = getFileContentStr(s"/order-history/$createDatabaseFileName")
+  private def getPostgresConnectionCfgString(serverName: String, port: Int): String =
+    s"""
+       |postgres {
+       |  server-name = $serverName
+       |  port-number = $port
+       |  database = $customDB
+       |  user = $customUser
+       |  password = $customPassword
+       |  data-source-class-name = "org.postgresql.ds.PGSimpleDataSource"
+       |}
+    """.stripMargin
 
-    postgresContainerLauncher.writeFile(
-      to = Paths.get(s"/docker-entrypoint-initdb.d/$createDatabaseFileName"),
-      content = createDatabaseFileContent
-    )
-  }
+  private val postgres: PostgreSQLContainer =
```

Comment:
maybe we should move this code to the base suite and add the trait with the DB cleanup? or move it to separate class and when we need the postgress in some test, write an annotation

Reply:
We use Postgres only in this test suite. We don't need ad-hoc trait with Postgres I guess